content/gsoc/2024/_index.md

```diff
@@ -3,22 +3,25 @@ title: PyBaMM GSoC 2024 Project Ideas
 ---
 
-## Migrate from `unittest` to `pytest`
+[↩ Go back to GSoC Home](/gsoc/)
+
+## Migrate from `unittest` to `pytest` and improve PyBaMM's testing infrastructure
 
 PyBaMM's inception predates the rise in popularity of `pytest`, and so we have used the `unittest` testing framework for our tests. However, `pytest` is now the de facto standard for testing in Python, and it is time to migrate our test cases over to it. The reason for this is that `pytest` is much more flexible and powerful than `unittest`, and will allow us to write better tests with less code. This may also involve migrating our test cases to use `hypothesis` for property-based testing, which will allow us to write even more powerful tests via the use of strategies, which are like generators for test cases.
 
-PyBaMM already uses `pytest` with the `nbmake` package to run our example notebooks as tests, but still uses `unittest` for the rest of the tests. `pytest-xdist` is also used to run the tests in parallel, which should end up being faster than the current serial implementation and should be adapted to be used with the new test framework.
+PyBaMM already uses `pytest` with the `nbmake` package to run our example notebooks as tests, but still uses `unittest` for the rest of the tests. See ([pybamm-team/PyBaMM #3617](https://github.com/pybamm-team/PyBaMM/issues/3617), which shall be one of the precursors to this project). `pytest-xdist` is also used in order to run the tests in parallel, which should end up being faster than the current serial implementation and should be adapted to be used with the new test framework and test cases.
 
-The student will be expected to look at other popular Python packages to see how they use `pytest` and `hypothesis` to write tests, and then migrate our tests over to use these frameworks. This will involve writing new tests, and also migrating existing tests over to the new framework, making use of and exploring the `pytest` documentation for scouting potential features to be incorporated. The student will also be expected to improve the test coverage of the codebase, and to write tests for any new code that they end up writing.
+The student will be expected to look at other popular Python packages to see how they use `pytest` and `hypothesis` to write tests, and then migrate our tests over to use these frameworks. This will involve writing new tests, and also migrating existing tests over to the new framework, making use of and exploring the `pytest` documentation for scouting potential features to be incorporated. The student will also be expected to improve the test coverage of the codebase, and to write tests for any new code that they end up writing. One of the niceties about `pytest` is its extensively configurable nature via `pytest` plugins, for which relevant ones can be scouted to be incoporated as a part of the migration process for unit and integration tests, coverage, doctests, and more.
```

Comment:
```suggestion
The student will be expected to look at other popular Python packages to see how they use `pytest` and `hypothesis` to write tests, and then migrate our tests over to use these frameworks. This will involve writing new tests, and also migrating existing tests over to the new framework, making use of and exploring the `pytest` documentation for scouting potential features to be incorporated. The student will also be expected to improve the test coverage of the codebase, and to write tests for any new code that they end up writing. One of the niceties about `pytest` is its extensively configurable nature via `pytest` plugins, for which relevant ones can be scouted to be incorporated as a part of the migration process for unit and integration tests, coverage, doctests, and more.
```